metals/src/main/scala/scala/meta/internal/metals/codeactions/InlineValueCodeAction.scala

```diff
@@ -0,0 +1,180 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.Defn
+import scala.meta.Pat
+import scala.meta.Template
+import scala.meta.Term
+import scala.meta.Tree
+import scala.meta.internal.metals.Compilers
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ServerCommands
+import scala.meta.internal.metals.clients.language.MetalsLanguageClient
+import scala.meta.internal.metals.codeactions.CodeAction
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+
+import org.eclipse.{lsp4j => l}
+
+class InlineValueCodeAction(
+    trees: Trees,
+    compilers: Compilers,
+    languageClient: MetalsLanguageClient,
+) extends CodeAction {
+
+  type CommandData = ServerCommands.InlineValueParams
+
+  override def command: Option[ActionCommand] =
+    Some(ServerCommands.InlineValue)
+
+  override def kind: String = l.CodeActionKind.RefactorInline
+
+  override def contribute(params: l.CodeActionParams, token: CancelToken)(
+      implicit ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    val pathStr = params.getTextDocument.getUri
+    val path = pathStr.toAbsolutePath
+    val range = params.getRange()
+    val action =
+      for {
+        termName <- getTermNameForPos(path, range)
+        if (hasDefinitionInScope(termName))
+        optThisDef = getDefinitionOfValue(path, range)
+        if (optThisDef.map(isLocal(_)).getOrElse(true))
+      } yield {
+        val command =
+          ServerCommands.InlineValue.toLsp(
+            ServerCommands.InlineValueParams(
+              params.getTextDocument(),
+              termName.pos.toLsp,
+              optThisDef.isDefined,
+            )
+          )
+        CodeActionBuilder.build(
+          title = InlineValueCodeAction.title(termName.value),
+          kind = this.kind,
+          command = Some(command),
+        )
+      }
+    action.toSeq
+  }
+
+  private def isLocal(definition: Defn.Val): Boolean =
+    definition.parent match {
+      case Some(_: Term.Block) => true
```

Comment:
Can you give an example I could include in the tests?

Reply:
For example:
```
def hello = {
   object O{
    val a = 123
   }
}
```
a is local in that case but would not be possible to inline.

Reply:
I thought we didn't want to inline `object` members? Although in this case I agree that we should allow it since it won't break anything in other files and it should a users decision to inline and remove this member. What about `class` members inside `def`?

Reply:
I think anything local is a safe bet the same as with rename.